spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java

```diff
@@ -369,6 +377,34 @@ public void testCustomExpressions() {
 		ctx.close();
 	}
 
+	@Test
+	public void testReconnect() throws Exception {
+		final MqttAsyncClient client = mock(MqttAsyncClient.class);
+		MqttPahoMessageDrivenChannelAdapter adapter = buildAdapter(client, null, ConsumerStopAction.UNSUBSCRIBE_NEVER);
+		adapter.setRecoveryInterval(10);
+		Log logger = spy(TestUtils.getPropertyValue(adapter, "logger", Log.class));
+		new DirectFieldAccessor(adapter).setPropertyValue("logger", logger);
+		given(logger.isDebugEnabled()).willReturn(true);
+		final AtomicInteger attemptingReconnectCount = new AtomicInteger();
+		willAnswer(i -> {
+			if (attemptingReconnectCount.getAndIncrement() == 0) {
+				adapter.connectionLost(new RuntimeException("while schedule running"));
+			}
+			i.callRealMethod();
+			return null;
+		}).given(logger).debug("Attempting reconnect");
+		ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+		taskScheduler.initialize();
+		adapter.setTaskScheduler(taskScheduler);
+		adapter.start();
+		adapter.connectionLost(new RuntimeException("initial"));
+		Thread.sleep(1000);
```

Comment:
No; the test is to make sure the reconnect is not performed more than twice. We need to wait a little bit to ensure that.

If you revert the src/main code, this test fails because the log is called many times.

Of course, now that it's no longer called on a schedule we can probably drop the test altogether.